src/vehiclelist.cpp

```diff
@@ -161,14 +166,19 @@ bool GenerateVehicleSortList(VehicleList *list, const VehicleListIdentifier &vli
 			break;
 
 		case VL_DEPOT_LIST:
-			for (const Vehicle *v : Vehicle::Iterate()) {
-				if (v->type == vli.vtype && v->IsPrimaryVehicle()) {
-					for (const Order *order : v->Orders()) {
-						if (order->IsType(OT_GOTO_DEPOT) && !(order->GetDepotActionType() & ODATFB_NEAREST_DEPOT) && order->GetDestination() == vli.index) {
-							list->push_back(v);
-							break;
-						}
+			for (const OrderList *orderlist : OrderList::Iterate()) {
+				const Vehicle *v = orderlist->GetFirstSharedVehicle();
+				if (v->type != vli.vtype) continue;
+
+				/* Vehicle is a candidate, search for a matching order. */
+				for (const Order *order = orderlist->GetFirstOrder(); order != nullptr; order = order->next) {
+					if (!(order->IsType(OT_GOTO_DEPOT) && !(order->GetDepotActionType() & ODATFB_NEAREST_DEPOT) && order->GetDestination() == vli.index)) continue;
```

Comment:
`if (!order->IsType(OT_GOTO_DEPOT) || (order->GetDepotActionType() & ODATFB_NEAREST_DEPOT) || order->GetDestination() != vli.index)) continue;` ? Could also be 3 if-statements. Not sure that matters.

Reply:
If makes it easier to read it's worth it.